### DISCUSSION TITLE
flask-security-too integrations with flask-smorest

### DISCUSSION THREAD

**Discussion by pejmanS21** [pallets-eco/flask-security#785] [Q&A] [answered]
Hello, I'm trying to integrate flask-smorest with flask-security-too for a password-based authentication app and after using`auth_required` decorator (or `login_required`) for a blueprint like following code, I've get these errors. can someone please help me with this?

[flask-smorest](https://github.com/marshmallow-code/flask-smorest)
[flask-security-too](https://github.com/Flask-Middleware/flask-security)

## Code

- `auth/views.py`
```python
from flask_security import auth_required, login_required, current_user
from flask_smorest import Blueprint, abort
from .usering.models.user import User
from .usering.models.role import Role
from .usering.models.user_role import UserRole

blp = Blueprint('auth', __name__, description='Endpoints for authentication')


@blp.route('/auth')
@auth_required()
class loginUser(MethodView):

    @blp.doc(description='Use this route to login as admin or user.')
    def get(self):
        pass

    @blp.doc(description='Use this route to login as admin or user.')
    def post(self):
        print(current_user.is_anonymous)
        return {"status": "OK"}
```

- `usering/models/user.py`

```python
class User(BaseDBModel, UserMixin):
    """The user table in the database."""

    __tablename__ = "user_account"
    id = Column(Integer(), primary_key=True)
    email = Column(String(255), unique=True)
    username = Column(String(255), unique=True, nullable=True)
    password = Column(String(255), nullable=False)
    fs_uniquifier = Column(String(255), unique=True, nullable=False)
    active = Column(Boolean())
    roles = relationship(
        "Role", secondary="user_role", backref=backref("users", lazy="dynamic")
    )
```

## Swagger 

![swagger](https://user-images.githubusercontent.com/73995528/234088382-5b686f93-3fbb-4f30-8a53-b44674968c83.png)

## Error

```
typeerror: The view function did not return a valid response. The return type must be a string, dict, list, tuple with headers or status, Response instance, or WSGI callable, but it was a loginUser.
```
![error](https://user-images.githubusercontent.com/73995528/234094821-17662448-5283-44f0-bc8c-3cfc9e9290d0.png)
![inspect](https://user-images.githubusercontent.com/73995528/234095559-bc6ca9da-aa42-492b-a040-402859a862f6.png)

@lafrech

**Comment by jwag956** [comment, marked as answer]
You can't decorate a class since that will cause the class itself to be returned (which is what the error is saying).
I think you are using Flask.views - in that case something like this should work:

`    class MyView(MethodView):
        decorators = [auth_required("token", "session")]

        def get(self):
            return render_template_string("Hi view")`

**Reply by pejmanS21** [reply]
Yes that's work correctly, I also use blueprint with it.

```python
view = loginUser.as_view("login_users")
view = auth_required(view)
blp.add_url_rule('/auth', view_func=view)
```

Thanks for your help @jwag956
